src/Systems/RenderSystem.h

```diff
@@ -1,28 +1,38 @@
 #pragma once
 
+#include <set>
 #include <vector>
+#include "Camera.h"
 #include "Config.h"
 #include "RenderComponent.h"
+#include "SFML/Graphics/RenderTexture.hpp"
 #include "SFML/Graphics/Sprite.hpp"
+#include "SFML/Graphics/VertexArray.hpp"
 #include "SFML/System/Vector2.hpp"
 #include "SFML/Window/Window.hpp"
 #include "System.h"
+#include "TransformComponent.h"
 
 namespace sf
 {
-    class RenderTexture;
     class RenderWindow;
-} // namespace sf
+}
 
 class RenderSystem : public System
 {
 public:
     RenderSystem();
     void update();
     void draw(sf::RenderTexture& window);
+    void clearMap();
 
 private:
     void displayPortal(Entity entity);
+    void renderMap(sf::RenderTexture& window, const std::vector<sf::Sprite*>& tiles, const Camera& camera);
+    void renderMap(sf::RenderTexture& window, const std::unordered_map<sf::Texture*, sf::VertexArray>& tiles,
+                   const Camera& camera);
+    void updateCamera(Camera& camera, sf::Vector2f targetPos, const sf::RenderTexture& window);
+    void updateQuad(Entity entity);
     void updatePlayerSprite(Entity entity);
```

Comment:
Const

Reply:
Dobra niby masz te consty w cpp, ale w h z jakiegoś powodu nie.

Reply:
Generalnie nie trzeba dodawać `const` do deklaracji jeśli typy argumentów definicji i deklaracji dalej się zgadzają (tutaj obie i tak działają na kopiach) - ale jest to bardzo zła praktyka, więc zmienione xD